File: sdks/typescript/examples/basic.ts
import { AxiomClient } from "../src/index";

async function run() {
  const client = new AxiomClient({
    baseUrl: "http://localhost:4500",
    keyName: "admin",
    keySecret: "YOUR_ADMIN_SECRET_KEY"
  });

  console.log("Checking databases...");
  const dbs = await client.listDatabases();
  console.log(JSON.stringify(dbs, null, 2));

  if (dbs.success && dbs.databases?.length) {
    const dbName = dbs.databases[0].name;

    console.log(`\nExecuting raw query on ${dbName}...`);
    const result = await client.query(dbName, "SELECT 1 as connected");
    console.log(JSON.stringify(result, null, 2));
  }
}

run();

File: sdks/python/examples/basic.py
from axiom import AxiomClient
import json

def main():
    client = AxiomClient(
        base_url="http://localhost:4500",
        key_name="admin",
        key_secret="YOUR_ADMIN_SECRET_KEY"
    )

    print("Checking databases...")
    dbs = client.list_databases()
    print(json.dumps(dbs, indent=2))

    if dbs.get("success") and dbs.get("databases"):
        db_name = dbs["databases"][0]["name"]
        
        print(f"\nExecuting raw query on {db_name}...")
        result = client.query(db_name, "SELECT 1 as connected")
        print(json.dumps(result, indent=2))

if __name__ == "__main__":
    main()

